momento-sdk/src/main/java/momento/sdk/responses/topic/SubscriptionState.java

```diff
@@ -0,0 +1,60 @@
+package momento.sdk.responses.topic;
+
+/** Represents the state of a subscription to a topic. */
+public class SubscriptionState {
+
+  private Runnable unsubscribeFn;
+  private Integer lastTopicSequenceNumber;
+  private boolean isSubscribed;
```

Comment:
I actually don't think that this boolean is needed. The only place it is used is

```
    if (isSubscribed) {
      unsubscribeFn.run();
      setUnsubscribed();
```

Instead we could do something like

```
```
  unsubscribeFn.run();
  this.unsubscribeFn = () -> {}
```

This would still ensure that each `unsubscribeFn` was only called once, but I think would meaningfully simplify this class

Reply:
The boolean is also used to mark the subscription to true when subscribing using the grpc client